recipes/grpc/all/conanfile.py

```diff
@@ -79,15 +79,23 @@ def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
             tools.check_min_cppstd(self, 11)
 
-        if tools.is_apple_os(self.settings.os) and "arm" in self.settings.arch:
+        # Temporary workaround because the conan center CI/CD can't handle M1 Macs.
+        if hasattr(self, "settings_build") and tools.cross_building(self):
+            # If cross bulding we check settings build
+            settings = self.settings_build
+        else:
+            # otherwise just settings
+            settings = self.settings
+
+        if tools.is_apple_os(settings.os) and "arm" in settings.arch:
             raise ConanInvalidConfiguration("gRPC is not supported on M1 Mac due to limitations in protobuf support.")
 
     def build_requirements(self):
         if hasattr(self, "settings_build"):
             self.build_requires('protobuf/3.17.1')
             # when cross compiling we need pre compiled grpc plugins for protoc
             if tools.cross_building(self):
-                self.build_requires('grpc/{}'.format(self.version))
+                self.build_requires('grpc/{}@{}/{}'.format(self.version, self.user, self.channel))
```

Comment:
This change is a bit unrelated and I'm not 100% sure what happens if user/channel is empty?
This is helpful when a custom user/channel is used (for example during testing).